src/Bounds.cpp

```diff
@@ -1235,8 +1235,26 @@ class Bounds : public IRVisitor {
                                        !b_interval.min.type().is_uint() &&
                                        can_prove(b_interval.min < 0 &&
                                                  b_interval.min > -t.bits())) {
-                                interval.min = a_interval.min >> abs(b_interval.min);
+                                if (a_interval.min.type().is_uint() || can_prove(a_interval.min >= 0)) {
+                                    interval.min = a_interval.min >> abs(b_interval.min);
+                                } else if (can_prove(a_interval.min < 0)) {
+                                    if (b_interval.has_upper_bound()) {
```

Comment:
I'm a bit puzzled by b_interval.min never shows up in here. We know it has a lower_bound because we're nested inside an if that requires it

Reply:
You're totally right, I don't think this optimization requires that `b` is lower bounded at all. Let me verify that to be sure first.

Reply:
Verified. Fixed in e8bcdf2